numalogic/models/autoencoder/variants/icvanilla.py

```diff
@@ -0,0 +1,196 @@
+from collections.abc import Sequence
+
+import torch
+from torch import nn, Tensor
+
+from numalogic.models.autoencoder.base import BaseAE
+from numalogic.tools.exceptions import LayerSizeMismatchError
+from numalogic.tools.layer import MultiChannelLinear
+
+
+class _VanillaEncoder(nn.Module):
+    r"""Encoder module for the VanillaAE.
+
+    Args:
+    ----
+        seq_len: sequence length / window length
+        n_features: num of features
+        layersizes: encoder layer size
+        dropout_p: the dropout value
+
+    """
+
+    def __init__(
+        self,
+        seq_len: int,
+        n_features: int,
```

Comment:
Have to decide if to go with n_features or n_channels name and have it consistent with the MultichannelAE

Reply:
Changed.